Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Services/LambdaRuntimeAPI.cs

```diff
@@ -39,18 +35,51 @@ public LambdaRuntimeApi(WebApplication app, IRuntimeApiDataStore runtimeApiDataS
         app.MapPost("/{functionName}/2018-06-01/runtime/invocation/{awsRequestId}/error", PostError);
     }
 
-    public Task<IResult> PostEventDefaultFunction(HttpContext ctx)
+    public static void SetupLambdaRuntimeApiEndpoints(WebApplication app)
+    {
+        _ = new LambdaRuntimeApi(app);
+    }
+
+    public Task PostEventDefaultFunction(HttpContext ctx)
     {
         return PostEvent(ctx, DefaultFunctionName);
     }
 
-    public async Task<IResult> PostEvent(HttpContext ctx, string functionName)
+    public async Task PostEvent(HttpContext ctx, string functionName)
     {
+        var runtimeDataStore = _runtimeApiDataStoreManager.GetLambdaRuntimeDataStore(functionName);
+
+        // RequestResponse mode is the default when invoking with the Lambda service client.
+        var isRequestResponseMode = true;
+        if (ctx.Request.Headers.TryGetValue("X-Amz-Invocation-Type", out var invocationType))
+        {
+            isRequestResponseMode = string.Equals(invocationType, "RequestResponse", StringComparison.InvariantCulture);
+        }
+
         using var reader = new StreamReader(ctx.Request.Body);
         var testEvent = await reader.ReadToEndAsync();
-        _runtimeApiDataStore.QueueEvent(testEvent);
+        var evnt = runtimeDataStore.QueueEvent(testEvent, isRequestResponseMode);
 
-        return Results.Accepted();
+        if (isRequestResponseMode)
+        {
+            evnt.WaitForCompletion();
+            var result = Results.Ok(evnt.Response);
+            ctx.Response.StatusCode = 200;
+
+            if (!string.IsNullOrEmpty(evnt.Response))
+            {
+                var responseData = Encoding.UTF8.GetBytes(evnt.Response);
```

Comment:
I removed the use of using `IResult` because they were adding another layer of JSON encoding that was altering the JSON sent by the Lambda function.